arangod/IResearch/IResearchDocument.cpp

```diff
@@ -514,6 +427,124 @@ void FieldIterator::reset(
   }
 }
 
+void FieldIterator::setBoolValue(VPackSlice const value) {
+  TRI_ASSERT(value.isBool());
+
+  auto& name = nameBuffer();
+
+  // mangle name
+  arangodb::iresearch::kludge::mangleBool(name);
+
+  // init stream
+  auto stream = BoolStreamPool.emplace();
+  stream->reset(value.getBool());
+
+  // set field properties
+  _value._name = name;
+  _value._analyzer = stream.release(); // FIXME don't use shared_ptr
+  _value._features = &irs::flags::empty_instance();
+}
+
+void FieldIterator::setNumericValue(VPackSlice const value) {
+  TRI_ASSERT(value.isNumber());
+
+  auto& name = nameBuffer();
+
+  // mangle name
+  arangodb::iresearch::kludge::mangleNumeric(name);
+
+  // init stream
+  auto stream = NumericStreamPool.emplace();
+  stream->reset(value.getNumber<double>());
```

Comment:
this may cause precision loss if the value is a 64 bit Int or UInt64.
(can be deleted via `VPackSlice::isInt()` or `VPackSlice::isUInt()`)

Reply:
at the moment all number are stored in ArangoSearch for 2 reasons:
- it's impossible to differentiate integers and floats in AqlValue
- range lookup will become very complicated

Reply:
AqlValues are capable of storing VPack values, which can distinguish between integers and floats.